osidb/tests/factories.py

```diff
@@ -64,20 +64,35 @@ class Meta:
     )
     statement = factory.LazyAttribute(lambda c: f"Statement for {c.cve_id}")
     embargoed = factory.Faker("random_element", elements=[False, True])
-    major_incident_state = factory.Maybe(
-        "is_major_incident",
-        yes_declaration=Flaw.FlawMajorIncident.APPROVED,
-        no_declaration=Flaw.FlawMajorIncident.NOVALUE,
+    major_incident_state = factory.Faker(
+        "random_element",
+        elements=[
+            Flaw.FlawMajorIncident.REQUESTED,
+            Flaw.FlawMajorIncident.REJECTED,
+            Flaw.FlawMajorIncident.APPROVED,
+            Flaw.FlawMajorIncident.CISA_APPROVED,
+            Flaw.FlawMajorIncident.NOVALUE,
+        ],
     )
-    summary = factory.Maybe(
-        "is_major_incident",
-        yes_declaration="I'm a spooky CVE",
-        no_declaration=factory.Faker("random_element", elements=["", "foo"]),
+    summary = factory.LazyAttribute(
+        lambda f: "I am a spooky CVE"
+        if f.major_incident_state
+        in [
+            Flaw.FlawMajorIncident.REQUESTED,
+            Flaw.FlawMajorIncident.APPROVED,
+            Flaw.FlawMajorIncident.CISA_APPROVED,
+        ]
+        else factory.Faker("random_element", elements=["", "foo"])
```

Comment:
I did some digging I am not sure this will work, when the `factory.Faker` is used inside the `LazyAttribute` it won't evaluate itself and the faker object itself will be assigned instead https://github.com/FactoryBoy/factory_boy/issues/915.

I've found a way how to resolve that, you need to use `class Params` inside the factory like this:

```python
class Params:
        fallback_summary = factory.Faker(
            "random_element", elements=["","foo"]
        )
...
summary = factory.LazyAttribute(
        lambda f: "I am a spooky CVE"
        if f.major_incident_state
        in [
            Flaw.FlawMajorIncident.REQUESTED,
            Flaw.FlawMajorIncident.APPROVED,
            Flaw.FlawMajorIncident.CISA_APPROVED,
        ]
        else f.fallback_summary
)          
```

or you can keep using the `factory.Maybe` and evaluating the complex condition in the `Params` class like this:

```python

class Params:
    complex_condition = factory.LazyAttribute(lambda f: True if f.major_incident_state in [...] else False)
```

and then use the `complex_condition` for the first argument of `factory.Maybe` and keep the `yes/no_declaration`.

Reply:
Thanks! Updated.